frontend/src/app/Settings/Users/UserDetail.tsx

```diff
@@ -106,39 +106,249 @@ const UserDetail: React.FC<Props> = ({
 
   const isUserSelf = () => user.id === loggedInUser.id;
 
-  let statusText;
-  switch (user.status) {
-    case "ACTIVE":
-      statusText = (
-        <span className="top-user-status padding-left-0">
-          {capitalizeText(user.role || "")}
+  function getUserStatusText() {
+    switch (user.status) {
```

Comment:
The userStatus (ACTIVE, PENDING, PROVISIONED) should be created somewhere as a constant IMO maybe place it in a util file so it can be reused outside this component even.